### DISCUSSION TITLE
Fails to deploy after successful build.

### DISCUSSION THREAD

**Discussion by vishalbasnet23** [vercel/next.js#44044] [Help]
Hi,
I am trying to deploy my application on Vercel however even though it gets build successfully on my server as well as in the deployment log, it fails and exits with this message:

```
info  - Checking validity of types...
--
13:33:36.374 | info  - Creating an optimized production build...
13:33:57.844 | info  - Compiled successfully
13:33:57.844 | info  - Collecting page data...
13:33:58.366 | node:internal/process/promises:288
13:33:58.367 | triggerUncaughtException(err, true /* fromPromise */);
13:33:58.368 | ^
13:33:58.369 |  
13:33:58.369 | [Error [Options Error]: url is required] { type: 'OptionsException' }
13:33:58.369 |  
13:33:58.370 | Node.js v18.12.1
13:33:58.433 | Error: Command "CI='' npm run build" exited with 1

```
Please let me know if you have any idea of what I might be doing wrong here.

Thank you in advance.

**Comment by icyJoseph** [comment]
Alright, so after Node 15+, uncaught promise rejections crash the process.

What version are you using in your development machine?

I'd also like to know if you are using GetServerSideProps?

## Edit

OP needed to add env vars to the Vercel dashboard.

**Reply by vishalbasnet23** [reply]
Thank you @icyJoseph, I am using the same version as the Vercel which is v. 18. And no I am not using GetServerSideProps.

**Reply by icyJoseph** [reply]
Alright, that makes thing even more strange. 

The error seems to suggest that a fetch operation, is missing a proper URL. Are you using environment variables? Are they correctly present in the Vercel Dashboard?

**Reply by vishalbasnet23** [reply]
Oh yeah, that might be the issue. Thank you @icyJoseph. I will configure the production env variables.

**Reply by icyJoseph** [reply]
Did it work? If so please don't forget to mark as answered.
